external-resolver/src/main/scala/org/genivi/sota/resolver/DeviceDataMigrator.scala

```diff
@@ -0,0 +1,44 @@
+package org.genivi.sota.resolver
+
+import java.time.Instant
+
+import org.genivi.sota.common.DeviceRegistry
+import org.genivi.sota.data.{PackageId, Uuid}
+import org.genivi.sota.resolver.db.ForeignPackages.InstalledForeignPackage
+import org.genivi.sota.resolver.db.{DeviceRepository, ForeignPackages}
+import org.slf4j.LoggerFactory
+import slick.driver.MySQLDriver.api._
+
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+object DeviceDataMigrator {
+
+  private lazy val logger = LoggerFactory.getLogger(this.getClass)
+
+  def apply(deviceRegistry: DeviceRegistry)(implicit db: Database, ec: ExecutionContext): Unit = {
+    db.run(packages()).onComplete {
```

Comment:
Using `onComplete` will install a callback on the future but we never know if it actually runs or runs partially, if the jvm quits before the callback finishes, for example, the migration could run partially. 

`apply` should either return a future, and then boot uses `Await` to wait for the result of the future, or we could use `Await` here, but we should not let this callback run orphan without any checking if it actually succeeds.